src/main/kotlin/command/Command.kt

```diff
@@ -327,14 +345,14 @@ abstract class Command {
                                     throw CommandExitException("Argument word " + it.getName() + " must not contain spaces (do not use quotes for this argument)")
                                 }
                             }
-                            if (it.getRequired() && parts.get(index).isEmpty()) {
+                            if (it.getRequired() && parts[index].isEmpty()) {
                                 if (it.getType() == ArgumentType.SUBCOMMAND) {
                                     throw CommandExitException("Argument subcommand " + it.getName() + " is required")
                                 } else {
                                     throw CommandExitException("Argument word " + it.getName() + " is required")
                                 }
                             }
-                            if (!it.getRequired() && parts.get(index).isEmpty()) {
+                            if (!it.getRequired() && parts[index].isEmpty()) {
```

Comment:
index should never be able to reach that point, as it is only incremented once per loop max, and the loop count is the same as the amount of items in the array. The only reason we use don't use a for in loop is because some types in the when() statement don't increment at all (such as argument.)